lib/dust.js

```diff
@@ -112,41 +112,79 @@
     return stream;
   };
 
+  /**
+   * Extracts a template function (body_0) from whatever is passed.
+   * @param nameOrTemplate {*} Could be:
+   *   - the name of a template to load from cache
+   *   - a CommonJS-compiled template (a function with a `template` property)
+   *   - a template function
+   * @param loadFromCache {Boolean} if false, don't look in the cache
+   * @return {Function} a template function, if found
+   */
+  function getTemplate(nameOrTemplate, loadFromCache/*=true*/) {
+    if(!nameOrTemplate) {
+      return;
+    }
+    if(typeof nameOrTemplate === 'function' && nameOrTemplate.template) {
+      // Sugar away CommonJS module templates
+      return nameOrTemplate.template;
+    }
+    if(dust.isTemplateFn(nameOrTemplate)) {
+      // Template functions passed directly
+      return nameOrTemplate;
+    }
+    if(loadFromCache !== false) {
+      // Try loading a template with this name from cache
+      return dust.cache[nameOrTemplate];
+    }
+  }
+
   function load(nameOrTemplate, chunk, context) {
     if(!nameOrTemplate) {
       return chunk.setError(new Error('No template or template name provided to render'));
     }
 
-    if(!dust.config.cache) {
+    if(dust.config.cache === false) {
```

Comment:
Interesting. Multiple caches is an interesting feature, though with onLoad able to return uncached functions, it can do its own caching at the loader layer. Might be better to expose that as the better answer.